core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -633,6 +634,21 @@ public Result visit(Project e) {
       if ((!isStar(e.getProjects(), e.getInput().getRowType(), e.getRowType())
           || style.isExpandProjection()) && !unpivotRelToSqlUtil.isStarInUnPivot(e, x)) {
         final List<SqlNode> selectList = new ArrayList<>();
+
+        List<String> aliases = new ArrayList<>();
+        if (x.node instanceof SqlSelect) {
+          SqlSelect sqlSelect = (SqlSelect) x.node;
+          if (sqlSelect.getFrom() instanceof SqlPivot) {
+            SqlPivot sqlPivot = (SqlPivot) sqlSelect.getFrom();
+            aliases = sqlPivot.inList.stream()
```

Comment:
extract to method
`pivotAliases = extractAliasesFromPivot(sqlPivot)`
```
/**
 * Extracts aliases from the inList of a SqlPivot node.
 *
 * @param sqlPivot The SqlPivot node.
 * @return A list of aliases extracted from the SqlPivot inList.
 */
private List<String> extractAliasesFromPivot(SqlPivot sqlPivot) {
    ..............................
}
```

Reply:
Done